### PR TITLE
TLS logging for session resumption 1.2

### PR DESCRIPTION
**Pull request by rtkrruvinskiy**
We assume session resumption has occurred if the Client Hello message included
a session id, we have not seen the server certificate, but we have seen a
Change Cipher Spec message from the server.

Previously, these transactions were not logged at all because the server cert
was never seen.

https://redmine.openinfosecfoundation.org/issues/1969

Replaces https://github.com/inliniac/suricata/pull/2536

### CODE REVIEW

**Comment by thus** [comment]
Prscript:
- PR thus-pcap: https://buildbot.openinfosecfoundation.org/builders/thus-pcap/builds/97
- PR thus: https://buildbot.openinfosecfoundation.org/builders/thus/builds/97

File: src/log-tlslog.c [hunk 1]
```diff
@@ -291,8 +297,11 @@ static int LogTlsLogger(ThreadVars *tv, void *thread_data, const Packet *p,
         return 0;
     }
 
-    if (ssl_state->server_connp.cert0_issuerdn == NULL ||
-            ssl_state->server_connp.cert0_subject == NULL) {
+    if ((ssl_state->server_connp.cert0_issuerdn == NULL ||
+            ssl_state->server_connp.cert0_subject == NULL) &&
+        ((ssl_state->flags & SSL_AL_FLAG_SESSION_RESUMED) == 0 ||
+         (hlog->flags & LOG_TLS_SESSION_RESUMPTION) == 0)) {
```
Comment (thus) [inline src/log-tlslog.c]: The indentation for the various parts of this "if" is not aligned with each other.

File: src/output-json-tls.c [hunk 2]
```diff
@@ -156,8 +164,10 @@ static int JsonTlsLogger(ThreadVars *tv, void *thread_data, const Packet *p,
         return 0;
     }
 
-    if (ssl_state->server_connp.cert0_issuerdn == NULL ||
-            ssl_state->server_connp.cert0_subject == NULL)
+    if ((ssl_state->server_connp.cert0_issuerdn == NULL ||
+            ssl_state->server_connp.cert0_subject == NULL) &&
+        ((ssl_state->flags & SSL_AL_FLAG_SESSION_RESUMED) == 0 ||
+         (tls_ctx->flags & LOG_TLS_SESSION_RESUMPTION) == 0))
```
Comment (thus) [inline src/output-json-tls.c]: Same as commented above.

**Comment by thus** [comment]
Except for the nitpick comments above, I think it looks good :)

**Comment by rtkrruvinskiy** [comment]
@thus In the identation, I aligned the parentheses of the two parts of the branch (because for me it makes it easy to see them), but I can easily change that. For clarity, how should I handle the indentation? In the part of the `if` that was already there, it looks like the second line is indented five spaces after the parenthesis on the first line. Should all subsequent lines also be indented at the same level as the second line?

**Comment by thus** [comment]
I personally find it more visually appealing to line up the following lines in an 'if' statement. The reason for using eight spaces (counted from the start of the 'if') is to avoid that any part of the 'if' statement aligns with the code coming right after, which (IMO) would make the code harder to read.

**Comment by rtkrruvinskiy** [comment]
Replaced by https://github.com/inliniac/suricata/pull/2562

**Pull request closed without merging**